### PR TITLE
[SPARK-42921][SQL][TESTS] Split `timestampNTZ/datetime-special.sql` into w/ and w/o `ansi` suffix to pass sql analyzer test in ansi mode

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
After https://github.com/apache/spark/pull/40496, run 

```
SPARK_ANSI_SQL_MODE=true build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
```

There is one test faild with `spark.sql.ansi.enabled = true`

```
[info] - timestampNTZ/datetime-special.sql_analyzer_test *** FAILED *** (11 milliseconds)
[info]   timestampNTZ/datetime-special.sql_analyzer_test
[info]   Expected "...date(999999, 3, 18, [false) AS make_date(999999, 3, 18)#x, make_date(-1, 1, 28, fals]e) AS make_date(-1, ...", but got "...date(999999, 3, 18, [true) AS make_date(999999, 3, 18)#x, make_date(-1, 1, 28, tru]e) AS make_date(-1, ..." Result did not match for query #1
[info]   select make_date(999999, 3, 18), make_date(-1, 1, 28) (SQLQueryTestSuite.scala:777)
[info]   org.scalatest.exceptions.TestFailedException:
```

The failure reason is the last parameter of function `MakeDate` is `failOnError: Boolean = SQLConf.get.ansiEnabled`.

So this pr split `timestampNTZ/datetime-special.sql` into w/ and w/o ansi to mask this test difference.

### Why are the changes needed?
Make SQLQueryTestSuite test pass with `spark.sql.ansi.enabled = true`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Pass GitHub Actions
- Manual checked `SPARK_ANSI_SQL_MODE=true build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"`

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @dtenedor @HyukjinKwon FYI

**Comment by dtenedor** [comment]
I am so sorry to break the build again :| thanks for fixing it! It looks like we need separate regular and ANSI test cases now!

**Review by dtenedor — APPROVED** [review]

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala [hunk 1]
```diff
@@ -88,6 +88,8 @@ class ThriftServerQueryTestSuite extends SQLQueryTestSuite with SharedThriftServ
     "datetime-special.sql",
     "ansi/datetime-special.sql",
     "timestampNTZ/datetime-special.sql",
+    // SPARK-42921
+    "timestampNTZ/datetime-special-ansi.sql",
```
Comment (LuciferYang) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala:92]: ignore in `ThriftServerQueryTestSuite`, same reason as `timestampNTZ/datetime-special.sql`

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
Thanks @HyukjinKwon @dtenedor

**Pull request closed without merging**